compiler/passes/src/function_inlining/inline_program.rs

```diff
@@ -41,7 +43,9 @@ impl ProgramReconstructor for FunctionInliner<'_> {
         assert!(input.functions.is_empty(), "All functions in the program scope should have been processed.");
 
         // Note that this intentionally clears `self.reconstructed_functions` for the next program scope.
-        let functions = core::mem::take(&mut self.reconstructed_functions);
+        let functions = core::mem::take(
+            &mut self.reconstructed_functions.iter().map(|(symbol, function)| (*symbol, function.clone())).collect(),
+        );
```

Comment:
```suggestion
        let functions = core::mem::take(&mut self.reconstructed_functions).into_iter().map(|(symbol, function)| (*symbol, function.clone())).collect();
```
You might even be able to call `collect` without `map`.